Panic in span End() when created with NewSpan

When creating a span using the now deprecated NewSpan function, calling End() on it panics. Panic trace:
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x9ebf4e]

goroutine 39 [running]:
testing.tRunner.func1(0xc42017e3c0)
	/usr/local/go/src/testing/testing.go:742 +0x454
panic(0xfd56e0, 0x164dfc0)
	/usr/local/go/src/runtime/panic.go:505 +0x24a
go.opencensus.io/trace.(*Span).End.func1()
	[XXX]/go.opencensus.io/trace/trace.go:269 +0x3e
sync.(*Once).Do(0xc42051b208, 0xc4204b4c90)
	/usr/local/go/src/sync/once.go:44 +0xd2
go.opencensus.io/trace.(*Span).End(0xc42051b1d0)
	[XXX]/go/src/go.opencensus.io/trace/trace.go:268 +0xa0
```

The simplest way to fix this is probably to replace ```s.executionTracerSpanEnd()``` by ```if s.executionTracerSpanEnd != nil { s.executionTracerSpanEnd() }```.

This is a serious bug caused by #715 .

--- a/trace/trace.go
+++ b/trace/trace.go
@@ -45,7 +45,7 @@ type Span struct {
 	*spanStore
 	endOnce sync.Once
 
-	executionTracerSpanEnd func() // ends the execution tracer span
+	executionTracerTaskEnd func() // ends the execution tracer span
 }
 
 // IsRecordingEvents returns true if events are being recorded for this span.
@@ -165,8 +165,8 @@ func StartSpan(ctx context.Context, name string, o ...StartOption) (context.Cont
 	}
 	span := startSpanInternal(name, parent != SpanContext{}, parent, false, opts)
 
-	ctx, end := startExecutionTracerSpan(ctx, name)
-	span.executionTracerSpanEnd = end
+	ctx, end := startExecutionTracerTask(ctx, name)
+	span.executionTracerTaskEnd = end
 	return NewContext(ctx, span), span
 }
 
@@ -180,8 +180,8 @@ func StartSpanWithRemoteParent(ctx context.Context, name string, parent SpanCont
 		op(&opts)
 	}
 	span := startSpanInternal(name, parent != SpanContext{}, parent, true, opts)
-	ctx, end := startExecutionTracerSpan(ctx, name)
-	span.executionTracerSpanEnd = end
+	ctx, end := startExecutionTracerTask(ctx, name)
+	span.executionTracerTaskEnd = end
 	return NewContext(ctx, span), span
 }
 
@@ -266,7 +266,9 @@ func (s *Span) End() {
 		return
 	}
 	s.endOnce.Do(func() {
-		s.executionTracerSpanEnd()
+		if s.executionTracerTaskEnd != nil {
+			s.executionTracerTaskEnd()
+		}
 		// TODO: optimize to avoid this call if sd won't be used.
 		sd := s.makeSpanData()
 		sd.EndTime = internal.MonotonicEndTime(sd.StartTime)

--- a/trace/trace_go11.go
+++ b/trace/trace_go11.go
@@ -21,6 +21,6 @@ import (
 	t "runtime/trace"
 )
 
-func startExecutionTracerSpan(ctx context.Context, name string) (context.Context, func()) {
+func startExecutionTracerTask(ctx context.Context, name string) (context.Context, func()) {
 	return t.NewContext(ctx, name)
 }

--- a/trace/trace_nongo11.go
+++ b/trace/trace_nongo11.go
@@ -20,6 +20,6 @@ import (
 	"context"
 )
 
-func startExecutionTracerSpan(ctx context.Context, name string) (context.Context, func()) {
+func startExecutionTracerTask(ctx context.Context, name string) (context.Context, func()) {
 	return ctx, func() {}
 }

